src/addon/client.cpp

```diff
@@ -288,6 +275,11 @@ bool addons_client::delete_remote_addon(const std::string& id, std::string& resp
 	config request_buf, response_buf;
 	config& request_body = request_buf.add_child("delete");
 
+	// the passphrase isn't provided, prompt for it
+	if(cfg["passphrase"].empty()) {
+		gui2::dialogs::addon_auth::execute(cfg);
```

Comment:
If you want to exit at this point you need to check whether `execute` returns false.

Reply:
Updated.